### PR TITLE
Remove IAsyncConnectionFactory

### PR DESCRIPTION
**Pull request by danielmarbach**
## Proposed Changes

I propose we remove the `IAsyncConnectionFactory` interface for v7. It has served it's purpose to evolve the client but now I don't see a reason to keep it anymore

## Types of Changes

What types of changes does your code introduce to this project?
_Put an `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes issue #NNNN)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause an observable behavior change in existing systems)
- [ ] Documentation improvements (corrections, new content, etc)
- [ ] Cosmetic change (whitespace, formatting, etc)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating
the PR. If you're unsure about any of them, don't hesitate to ask on the
mailing list. We're here to help! This is simply a reminder of what we are
going to look for before merging your code._

- [x] I have read the `CONTRIBUTING.md` document
- [x] I have signed the CA (see https://cla.pivotal.io/sign/rabbitmq)
- [x] All tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in related repositories

## Further Comments

Should not be backported

### CODE REVIEW

**Review by michaelklishin — APPROVED** [review]

**Comment by danielmarbach** [comment]
@michaelklishin We have not yet established a formal rule how obsoletion should take place so I decided to just remove the interface. Of course we could also do a less drastic step of still having the interface around but obsolete with error. I'm mentioning it for completeness reasons

File: projects/RabbitMQ.Client/client/impl/Connection.cs [hunk 1]
```diff
@@ -102,14 +102,9 @@ public Connection(IConnectionFactory factory, bool insist, IFrameHandler frameHa
...
-            if (factory is IAsyncConnectionFactory asyncConnectionFactory && asyncConnectionFactory.DispatchConsumersAsync)
-            {
-                ConsumerWorkService = new AsyncConsumerWorkService(factory.ConsumerDispatchConcurrency);
-            }
-            else
-            {
-                ConsumerWorkService = new ConsumerWorkService(factory.ConsumerDispatchConcurrency);
-            }
+            ConsumerWorkService = factory.DispatchConsumersAsync
+                ? new AsyncConsumerWorkService(factory.ConsumerDispatchConcurrency)
```
Comment (stebet) [inline projects/RabbitMQ.Client/client/impl/Connection.cs:106]: I'm also wondering if there is a case to be made to simply remove the non-async consumers from 7.0 and keep them async-only?

**Review by stebet — APPROVED** [review]

File: projects/RabbitMQ.Client/client/impl/Connection.cs (hunk 1 above)
Comment (danielmarbach) [reply projects/RabbitMQ.Client/client/impl/Connection.cs:106]: I think that is one discussion topic I was hopping to have in the meeting :D, then write it down and announce it to the community

Comment (michaelklishin) [reply projects/RabbitMQ.Client/client/impl/Connection.cs:106]: We can consider it.

**Pull request merged**